apps/frontend/src/pages/projects/projects.tsx

```diff
@@ -47,27 +50,53 @@ const Projects = (): JSX.Element => {
 		projectPatchStatus,
 		projects,
 		projectStatus,
+		projectsTotalCount,
 	} = useAppSelector(({ projects }) => projects);
 
+	const handleSearchChange = useCallback(
+		(value: string) => {
+			onSearch(value);
+		},
+		[onSearch],
+	);
+
 	useEffect(() => {
 		if (projectToModifyId) {
 			void dispatch(projectActions.getById({ id: String(projectToModifyId) }));
 		}
 	}, [dispatch, projectToModifyId]);
 
+	const hasProjects = projects.length !== EMPTY_LENGTH;
+
+	const { hasNextPage, onPageChange, onPageReset, page, pageSize } =
+		useInfiniteScroll({
+			pageSize: PROJECTS_PAGE_SIZE,
+			totalItemsCount: projectsTotalCount,
+		});
+
 	useEffect(() => {
-		void dispatch(projectActions.loadAll(search));
-	}, [dispatch, search]);
+		void dispatch(
+			projectActions.loadAll({
+				name: search,
+				page,
+				pageSize,
+			}),
+		);
+	}, [dispatch, search, page, pageSize]);
```

Comment:
There is a problem with extra requests on search changing before page has changed. I have to remove search dependency to make it work properly, but I can't. So, I need some help with this

Reply:
Why can't you remove it? is it a linter error or something else?

Reply:
Yes, I get warning
```
React Hook useEffect has missing dependencies: 'dispatch', 'page', 'pageSize', and 'search'. Either include them or remove the dependency array
```

Reply:
let's use eslint disable next line, only for this case 🥲

Reply:
hm, got this, then looks like there should be another solution to this issue, maybe we need to return isIntersect from the intersection hook and then pass it to this useEffect?

Reply:
It works without the extra request if to change order in code

![image](https://github.com/user-attachments/assets/3c0384ea-314a-4ed1-afec-66c138e4a4d3)

Reply:
I have solution with adding `const [isLoaded, setIsLoaded] = useState<boolean>(false);` into `useInfiniteScroll` and set it `false` on page changing and `true` in loading dispatch:
```
	useEffect(() => {
		if (!isLoaded) {
			void dispatch(
				projectActions.loadAll({
					name: search,
					page,
					pageSize,
				}),
			);
			
			setIsLoaded(true)
		}
	}, [dispatch, page, pageSize, search]);
```

Reply:
Maybe we can just use isLoading from the statuses we have in slice?

Reply:
or pass the loading function into `useInfiniteScroll` as parameter on page changing:
```
	const loadProjects = useCallback((page: number, pageSize: number) => {
		void dispatch(
			projectActions.loadAll({
				name: search,
				page,
				pageSize,
			}),
		);
	}, [dispatch, search]);

	const { hasNextPage, onPageChange, onPageReset, isLoaded,
		setIsLoaded, page, pageSize } =
		useInfiniteScroll({
			pageSize: PROJECTS_PAGE_SIZE,
			totalItemsCount: projectsTotalCount,
			handleLoading: loadProjects,
		});
```

Reply:
then let's implement your suggestion above

Reply:
already did it